app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -71,6 +100,17 @@ private void storeStuct(IASTDeclaration[] declarations, String name) throws Exce
         structRepo.put(name, struct);
     }
 
+    public ArrayList getAssociatedStruct(String IDName) {
+        for (String struct : IDStruct.keySet()) {
+            PayloadMap curr = IDStruct.get(struct);
+            System.out.println("This is ID: " + IDName + " and Struct: " + struct);
+            if (curr.canIDNames.contains(IDName)) {
+                return getStruct(curr.struct);
+            }
+        }
+        return null;
```

Comment:
Can this whole thing be simplified? Instead of storing the `simplifyName` result for each CAN ID const in a dedicated data structure and searching through the entire structure, can you just run `simplifyName` on `IDName` and index into the IDStruct from there?

Note that iterating through a list is relatively expensive (since it needs to look for every element), compared to a hash table lookup (when you know exactly what key you're looking for). For this application, it practically won't matter (since the lists are small enough), but it's contrary to good practice.

Additionally, if you can simplify this, you might be able to get rid of PayloadMap completely, where `idStruct` is a direct simplified name to struct map.